Tests don't work -- missing files and no steps

Currently, the tests do not work:

```
xobs@Pazuzu:~/Code/Fomu/TinyFPGA-Bootloader/tests/win10_enumeration_test$ make
iverilog -I.. -s top_tb -o test -c ../file_list.txt
../top_tb_header.vh:32: error: Unknown module type: vlog_tb_utils
../top_tb_header.vh:42: error: Unknown module type: vlog_tap_generator
3 error(s) during elaboration.
*** These modules were missing:
        vlog_tap_generator referenced 1 times.
        vlog_tb_utils referenced 1 times.
***
../test.mk:2: recipe for target 'test' failed
make: *** [test] Error 3
xobs@Pazuzu:~/Code/Fomu/TinyFPGA-Bootloader/tests/win10_enumeration_test$
```

They appear to depend on a module called `vlog_tb_utils`, which isn't mentioned anywhere.

If the files are added (e.g. from https://github.com/fusesoc/vlog_tb_utils), or if those lines are removed, then the error goes away, however the timestep never advances:

```
xobs@Pazuzu:~/Code/Fomu/TinyFPGA-Bootloader/tests/win10_enumeration_test$ make
iverilog -I.. -s top_tb -o test.vvp -c ../file_list.txt
vvp -v test.vvp
Compiling VVP ...
 ... VVP file version 11.0 (devel) (s20150603-616-ge93ee39)
Compile cleanup...
 ... Linking
 ... Removing symbol tables
 ... Compiletf functions
 ...      596 functors (net_fun pool=524288 bytes)
                148 logic
                  0 bufif
                  0 resolv
                220 signals
 ...      533 filters (net_fil pool=524288 bytes)
 ...     6375 opcodes (172032 bytes)
 ...      501 nets
 ...      596 vvp_nets (1048544 bytes)
 ...        0 arrays (0 words)
 ...       12 memories
                 12 logic (190 words)
                  0 real (0 words)
 ...       63 scopes
 ... 0.03125 seconds, 25184.0/5300.0/2296.0 KBytes size/rss/shared
Running ...
 ...execute EndOfCompile callbacks
 ...propagate initialization events
 ...execute StartOfSim callbacks
 ...run scheduler
           ... WAIT ~60 SECONDS ...
^C** VVP Stop(0) **
** Flushing output streams.
** Current simulation time is 0 ticks.
> finish
** Continue **
 ...execute Postsim callbacks
 ... 66.9531 seconds, 26844.0/5900.0/2888.0 KBytes size/rss/shared
Event counts:
           3 time steps (pool=113)
    38707312 thread schedule events
    232244005 assign events
             ...assign(vec4) pool=9362
             ...assign(vec8) pool=204
             ...assign(real) pool=256
             ...assign(word) pool=128
             ...assign(word/r) pool=204
         302 other events (pool=4096)

xobs@Pazuzu:~/Code/Fomu/TinyFPGA-Bootloader/tests/win10_enumeration_test$
```

--- a/TinyFPGA-Bootloader.core
+++ b/TinyFPGA-Bootloader.core
@@ -4,6 +4,7 @@ name : ::TinyFPGA-Bootloader:0
 filesets:
   common:
     files:
+      - common/strobe.v
       - common/usb_fs_out_arb.v
       - common/usb_reset_det.v
       - common/usb_serial_ctrl_ep.v

--- a/common/strobe.v
+++ b/common/strobe.v
@@ -19,8 +19,8 @@ module strobe(
 	initial begin
 		flag = 0;
 		prev_strobe = 0;
-		sync = 0;
-		data = 0;
+		sync[DELAY:0] = 0;
+		data[WIDTH-1:0] = 0;
 	end
 
 	// flip the flag and clock in the data when strobe is high
@@ -55,7 +55,7 @@ module dflip(
 );
 	reg [2:0] d;
 	initial begin
-		d = 0;
+		d[2:0] = 0;
 	end
 	always @(posedge clk)
 		d <= { d[1:0], in };

--- a/common/strobe.v
+++ b/common/strobe.v
@@ -16,6 +16,13 @@ module strobe(
 	reg [DELAY:0] sync;
 	reg [WIDTH-1:0] data;
 
+	initial begin
+		flag = 0;
+		prev_strobe = 0;
+		sync = 0;
+		data = 0;
+	end
+
 	// flip the flag and clock in the data when strobe is high
 	always @(posedge clk_in) begin
 		//if ((strobe_in && !prev_strobe)
@@ -47,6 +54,9 @@ module dflip(
 	output out
 );
 	reg [2:0] d;
+	initial begin
+		d = 0;
+	end
 	always @(posedge clk)
 		d <= { d[1:0], in };
 	assign out = d[2];

--- a/common/usb_fs_out_pe.v
+++ b/common/usb_fs_out_pe.v
@@ -160,65 +160,65 @@ module usb_fs_out_pe #(
       end
       always @* begin
 
-        ep_state_next[ep_num] <= ep_state[ep_num];
+        ep_state_next[ep_num] = ep_state[ep_num];
 
         if (out_ep_stall[ep_num]) begin
-          ep_state_next[ep_num] <= STALL;
+          ep_state_next[ep_num] = STALL;
 
         end else begin
           case (ep_state[ep_num])
             READY_FOR_PKT : begin
               if (out_xfr_start && rx_endp == ep_num) begin
-                ep_state_next[ep_num] <= PUTTING_PKT;
+                ep_state_next[ep_num] = PUTTING_PKT;
 
               end else begin
-                ep_state_next[ep_num] <= READY_FOR_PKT;
+                ep_state_next[ep_num] = READY_FOR_PKT;
               end
             end
 
             PUTTING_PKT : begin
               if (new_pkt_end && current_endp == ep_num) begin
-                ep_state_next[ep_num] <= GETTING_PKT;
+                ep_state_next[ep_num] = GETTING_PKT;
 
               end else if (rollback_data && current_endp == ep_num) begin
-                ep_state_next[ep_num] <= READY_FOR_PKT;
+                ep_state_next[ep_num] = READY_FOR_PKT;
 
               end else begin
-                ep_state_next[ep_num] <= PUTTING_PKT;
+                ep_state_next[ep_num] = PUTTING_PKT;
               end
             end
 
             GETTING_PKT : begin
 
               if (ep_get_addr[ep_num][5:0] >= (ep_put_addr[ep_num][5:0] - 2)) begin
-                ep_state_next[ep_num] <= READY_FOR_PKT;
+                ep_state_next[ep_num] = READY_FOR_PKT;
 
               end else begin
-                ep_state_next[ep_num] <= GETTING_PKT;
+                ep_state_next[ep_num] = GETTING_PKT;
               end
             end
 
             STALL : begin
               if (setup_token_received && rx_endp == ep_num) begin
-                ep_state_next[ep_num] <= READY_FOR_PKT;
+                ep_state_next[ep_num] = READY_FOR_PKT;
 
               end else begin
-                ep_state_next[ep_num] <= STALL;
+                ep_state_next[ep_num] = STALL;
               end
             end
 
             default begin
-              ep_state_next[ep_num] <= READY_FOR_PKT;
+              ep_state_next[ep_num] = READY_FOR_PKT;
             end
           endcase
         end
 
         if (ep_state_next[ep_num][1:0] == READY_FOR_PKT) begin
-          ep_get_addr_next[ep_num][5:0] <= 0;
+          ep_get_addr_next[ep_num][5:0] = 0;
         end else if (ep_state_next[ep_num][1:0] == GETTING_PKT && out_ep_data_get[ep_num]) begin
-          ep_get_addr_next[ep_num][5:0] <= ep_get_addr[ep_num][5:0] + 1;
+          ep_get_addr_next[ep_num][5:0] = ep_get_addr[ep_num][5:0] + 1;
         end else begin
-          ep_get_addr_next[ep_num][5:0] <= ep_get_addr[ep_num][5:0];
+          ep_get_addr_next[ep_num][5:0] = ep_get_addr[ep_num][5:0];
         end
       end
 

--- a/common/usb_fs_out_pe.v
+++ b/common/usb_fs_out_pe.v
@@ -152,6 +152,12 @@ module usb_fs_out_pe #(
   genvar ep_num;
   generate
     for (ep_num = 0; ep_num < NUM_OUT_EPS; ep_num = ep_num + 1) begin
+      initial begin
+        ep_state_next[ep_num] = READY_FOR_PKT;
+        ep_state[ep_num] = READY_FOR_PKT;
+        ep_get_addr_next[ep_num] = 0;
+        ep_get_addr[ep_num] = 0;
+      end
       always @* begin
 
         ep_state_next[ep_num] <= ep_state[ep_num];

--- a/common/usb_fs_tx.v
+++ b/common/usb_fs_tx.v
@@ -44,6 +44,9 @@ module usb_fs_tx (
   // convert tx_data_get from 48 to clk
   //wire tx_data_get_48 = tx_data_get;
   reg tx_data_get_48;
+  initial begin
+    tx_data_get_48 = 0;
+  end
   strobe tx_data_get_strobe(
 	.clk_in(clk_48mhz),
 	.clk_out(clk),

--- a/tests/file_list.txt
+++ b/tests/file_list.txt
@@ -1,4 +1,6 @@
 test.v
+../vlog_tb_utils/vlog_tb_utils.v
+../vlog_tb_utils/vlog_tap_generator.v
 ../../common/edge_detect.v
 ../../common/strobe.v
 ../../common/tinyfpga_bootloader.v

